pkg/operator/k8sutil/deployment.go

```diff
@@ -80,6 +84,15 @@ func UpdateDeploymentAndWait(context *clusterd.Context, deployment *apps.Deploym
 
 	// wait for the deployment to be restarted
 	sleepTime := 2
```

Comment:
Instead of using an env var you should be able to move this declaration up to line 70 before the call to `util.Retry()`, then set it directly. Just make sure not to re-declare the variable.

Reply:
Actually, I don't think changing this sleep time will make a noticeable difference. There could still be a timing issue that the OSD won't be listening for a few seconds even after the pod is running. Instead, what is more likely to help (until we have a proper liveness probe) is adding a sleep after the pod is in running state. For example, add a sleep before returning on line 116 below. But that would affect all daemons, not just the OSDs. But even that is just a hack. 

Instead of skipping the upgrade checks, it's still much better to wait for the ok-to-stop for some time even if we continue. It feels like a setting is needed in the cluster CR such as `continueUpgradeAfterChecksEvenIfNotClean` so the orchestration can continue with the upgrade when an admin is not around to troubleshoot.

Reply:
Yes, the checks still make sense to me as well, they will still avoid issues. My issue is to add yet another CR option... But I don't any better way at the moment.